riptable/rt_struct.py

```diff
@@ -4335,90 +4335,106 @@ def __eq__(self, other):
+                        col_idwrapper = IdWrapper(array_data)
+
+                        # Always add this object's size to the total logical size of the Struct data
+                        # (without regard to whether this is a duplicate/aliased object).
+                        total_logical_size += array_data.nbytes       # assumes data is an np.ndarray or a subclass of it
+
+                        # If the data hasn't been seen yet (i.e. it's not an alias of some other data we've seen before)
+                        # add it to the "seen objects" set and add it's size to the total physical size.
+                        if col_idwrapper not in seen_objects:
+                            seen_objects.add(col_idwrapper)
+                            total_physical_size += array_data.nbytes
+
+                    elif isinstance(obj, np.ndarray):
+                        # Handle array views by searching until we find the "root" base array.
+                        # An ndarray is a view if it returns a non-None instance from it's .base property.
+                        while obj.base is not None:
+                            obj = obj.base
+
+                        # Create an IdWrapper around the column data.
+                        col_idwrapper = IdWrapper(obj)
+
+                        # Always add this object's size to the total logical size of the Struct data
+                        # (without regard to whether this is a duplicate/aliased object).
+                        total_logical_size += obj.nbytes       # assumes data is an np.ndarray or a subclass of it
+
+                        # If the data hasn't been seen yet (i.e. it's not an alias of some other data we've seen before)
+                        # add it to the "seen objects" set and add it's size to the total physical size.
+                        if col_idwrapper not in seen_objects:
+                            seen_objects.add(col_idwrapper)
+                            total_physical_size += obj.nbytes
+
+                    # If the current item is a Struct, recurse downwards.
+                    # This _must_ be the last case before the fallthrough so that any classes
+                    # derived from Struct are handled in some more-specific way before taking
+                    # the general case for Struct.
+                    elif isinstance(obj, Struct):
+                        # Create an IdWrapper around the column data.
+                        col_idwrapper = IdWrapper(obj)
+
+                        # If the data hasn't been seen yet (i.e. it's not an alias of some other data we've seen before)
+                        # add it to the "seen objects" set and add to the list of pending Struct instances to process.
+                        if col_idwrapper not in seen_objects:
+                            seen_objects.add(col_idwrapper)
+                            pending_objects.append(obj)

```

Comment:
Why don't we just call `obj.total_sizes` and add the physical and logical sizes to the totals? I think we don't need the `pending_objects` queue.